centreon/src/Core/Service/Infrastructure/Repository/DbWriteServiceActionLogRepository.php

```diff
@@ -0,0 +1,266 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Service\Infrastructure\Repository;
+
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Domain\Repository\RepositoryException;
+use Centreon\Infrastructure\DatabaseConnection;
+use Core\ActionLog\Application\Repository\WriteActionLogRepositoryInterface;
+use Core\ActionLog\Domain\Model\ActionLog;
+use Core\Common\Application\Converter\YesNoDefaultConverter;
+use Core\Common\Domain\YesNoDefault;
+use Core\Common\Infrastructure\Repository\AbstractRepositoryRDB;
+use Core\Domain\Common\GeoCoords;
+use Core\Service\Application\Repository\ReadServiceRepositoryInterface;
+use Core\Service\Application\Repository\WriteServiceRepositoryInterface;
+use Core\Service\Domain\Model\NewService;
+use Core\Service\Domain\Model\NotificationType;
+use Core\Service\Domain\Model\Service;
+use Core\Service\Infrastructure\Model\NotificationTypeConverter;
+
+class DbWriteServiceActionLogRepository extends AbstractRepositoryRDB implements WriteServiceRepositoryInterface
+{
+    use LoggerTrait;
+    public const SERVICE_OBJECT_TYPE = 'service';
+
+    /**
+     * @param WriteServiceRepositoryInterface $writeServiceRepository
+     * @param ContactInterface $contact
+     * @param ReadServiceRepositoryInterface $readServiceRepository
+     * @param WriteActionLogRepositoryInterface $writeActionLogRepository
+     * @param DatabaseConnection $db
+     */
+    public function __construct(
+        private readonly WriteServiceRepositoryInterface $writeServiceRepository,
+        private readonly ContactInterface $contact,
+        private readonly ReadServiceRepositoryInterface $readServiceRepository,
+        private readonly WriteActionLogRepositoryInterface $writeActionLogRepository,
+        DatabaseConnection $db
+    ) {
+        $this->db = $db;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function delete(int $serviceId): void
+    {
+        try {
+            $service = $this->readServiceRepository->findById($serviceId);
+            if ($service === null) {
+                throw new RepositoryException(sprintf('Cannot find service to delete (ID: %d).', $serviceId));
+            }
+            
+            $this->writeServiceRepository->delete($serviceId);
+
+            $actionLog = new ActionLog(
+                self::SERVICE_OBJECT_TYPE,
+                $serviceId,
+                $service->getName(),
+                ActionLog::ACTION_TYPE_DELETE,
+                $this->contact->getId()
+            );
+            $this->writeActionLogRepository->addAction($actionLog);
+        } catch (\Throwable $ex) {
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+
+            throw $ex;
```

Comment:
Why you doesn't use RepositoryException instead of throw an other exception ? It would be clearer for the error handling.

Reply:
As you want, it's your opinion even if I don't agree :) I resolve ;)